presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -167,9 +173,31 @@ protected Boolean visitAtTimeZone(AtTimeZone node, Void context)
         @Override
         protected Boolean visitSubqueryExpression(SubqueryExpression node, Void context)
         {
+            AstUtils.preOrder(node)
+                    .filter(isInstanceOfAny(DereferenceExpression.class, Identifier.class))
+                    .map(Expression.class::cast)
+                    .forEach(this::analyzeSubqueryReference);
+
             return true;
         }
 
+        private void analyzeSubqueryReference(Expression referenceExpression)
+        {
+            ResolvedField resolvedField = analysis.getScope(referenceExpression)
```

Comment:
Oh I see, it works because of `GetScopeVisitor`, but it might return output scope of wrapping node (e.g: output scope of `QuerySpecificaction` for an select expression), which you shouldn't resolve select symbol against.

For instance, `select (a + 1) as a from ...`. This code would resolve expression `a` from `a+1` against output scope, so it will basically resolve back to `a+1`. Even if this works in the case of this check, this is fishy at least. 

I think analysis should store `ResolvedField` explicitly for every `DefererenceExpression`, `Identity` and `FieldReference`. It should be filled by `ExpressionAnalyzer`. Then it would be easy to check if reference points to grouping fields from correct scope. I have similar problem in my ongoing work for implicit coercions for ORDER BY expressions. I am going to solve the issue as described.